include/mbedtls/ccm.h

```diff
@@ -40,6 +42,106 @@ typedef struct {
 }
 mbedtls_ccm_context;
 
+#else /* MBEDTLS_CCM_ALT */
+#include "ccm_alt.h"
+
+typedef struct {
+    union
+    {
+        mbedtls_ccm_alt_context_t ccm_alt_ctx;    /*!< opaque context structure to be defined in ccm_alt.h */
+        mbedtls_cipher_context_t cipher_ctx;
+    }u_ctx;
+    unsigned int    is_alternative;
```

Comment:
ok. The idea was to put the context in the beginning, for casting purposes, but I will change this